lib/functions/buildFileList.sh

```diff
@@ -428,7 +428,10 @@ function BuildFileList() {
     ########################
     # Use BASE_FILE here because FILE_TYPE is not reliable when there is no file extension
     elif [[ "${FILE_TYPE}" != "tap" ]] && [[ "${FILE_TYPE}" != "yml" ]] &&
-      [[ "${FILE_TYPE}" != "yaml" ]] && [[ "${FILE_TYPE}" != "json" ]] && [[ "${FILE_TYPE}" != "xml" ]] && [[ "${BASE_FILE}" =~ .*(contain|dock)erfile.* ]]; then
+      [[ "${FILE_TYPE}" != "yaml" ]] && [[ "${FILE_TYPE}" != "json" ]] &&
+      [[ "${FILE_TYPE}" != "xml" ]] &&
+      [[ "${BASE_FILE}" =~ "^(.+\.)?(contain|dock)erfile$" ]]; then
```

Comment:
```suggestion
      [[ "${BASE_FILE}" =~ ^(.+\.)?(contain|dock)erfile$ ]]; then
```

To fix, https://github.com/super-linter/super-linter/actions/runs/6858363401/job/18648985531?pr=4867#step:7:1858